plugins/outputs/opensearch/README.md

```diff
@@ -0,0 +1,371 @@
+# OpenSearch Output Plugin
+
+This plugin writes to [OpenSearch](https://opensearch.org/) via HTTP
+
+It supports OpenSearch releases from 1 and 2. Future comparability with 1.x is
+not guaranteed and instead will focus on 2.x support. Consider using the
+existing Elasticsearch plugin for 1.x.
+
+## Global configuration options <!-- @/docs/includes/plugin_config.md -->
+
+In addition to the plugin-specific configuration settings, plugins support
+additional global and plugin configuration settings. These settings are used to
+modify metrics, tags, and field or create aliases and configure ordering, etc.
+See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
+
+[CONFIGURATION.md]: ../../../docs/CONFIGURATION.md#plugins
+
+## Configuration
+
+```toml @sample.conf
+# Configuration for OpenSearch to send metrics to.
+[[outputs.OpenSearch]]
+  ## URLs
+  ## The full HTTP endpoint URL for your OpenSearch instance. Multiple URLs can
+  ## be specified as part of the same cluster, but only one URLs is used to
+  ## write during each interval.
+  urls = ["http://node1.os.example.com:9200"]
+
+  ## Index Name
+  ## The target index for metrics (OpenSearch will create if it not exists).
+  ## You can use the date specifiers below to create indexes per time frame.
+  ## The metric timestamp will be used to decide the destination index name
+  ##   %Y - year (2016)
+  ##   %y - last two digits of year (00..99)
+  ##   %m - month (01..12)
+  ##   %d - day of month (e.g., 01)
+  ##   %H - hour (00..23)
+  ##   %V - week of the year (ISO week) (01..53)
+  ## For example: telegraf-%Y.%m.%d would set it to telegraf-2006-01-02
+  index_name = ""
+
```

Comment:
The idea of @Hipska is to reuse go templates so this should be
```suggestion
  ## Target index name for metrics (OpenSearch will create if it not exists).
  ## This is a Golang template (see https://pkg.go.dev/text/template) to
  ## access the metric name (`{{.Name}}`), a tag value (`{{.Tag "name"}}`) ,
  ## a field value (`{{.Field "name"}}`) or the timestamp (`{{.Time}}`).
  ## For example: "telegraf-{{.Time.Format "2006-01-02"}}" would set it to telegraf-2023-07-27
  index_name = ""

```

Reply:
Used with little change as {{Name}}`), (`{{Tag "name"}}`), (`{{Field "name"}}`) are not prefixed with a .
because of the way string-template's implementation wrt. function.

Reply:
If using `templateMetric`, then these methods come available to the object. This is a crucial part.

Reply:
made the change, working as expected.